### PR TITLE
OT-494 & OT-495: Make web links clickable & Simple Markdown

### LINKED ISSUE #247: Make web links clickable
**Is your feature request related to a problem? Please describe.**
When I receive or send a web link, I cannot open the link, I can only copy the whole message.

**Describe the solution you'd like**
I want to be able to click/tap on the link to open it.

[Internal bug tracker](https://jira.open-xchange.com/browse/OT-494)

### LINKED ISSUE #248: Allow simple markup tags
**Is your feature request related to a problem? Please describe.**
It's hard to structure text without formatting.

**Describe the solution you'd like**
It would be great to support the following markup:
```
**bold**
*italic*
_underlined_
* unordered list item 1
* unordered list item 2
1. ordered list item 1
2. ordered list item 2
>  quote
`code`
```

**Describe alternatives you've considered**
We could use HTML

[Internal bug tracker](https://jira.open-xchange.com/browse/OT-495)

### PR DESCRIPTION
**Pull request by phranck**
**Link to the given issue**
Fixes #247 - Make web links clickable
Fixes #248 - Allow simple markup tags

**What was done?**
* Adds two new plugins
  * [flutter_markdown](https://pub.dev/packages/flutter_markdown)
  * [html2md](https://pub.dev/packages/html2md)
* Renames `utils/text.dart` to `extensions/string_apis.dart`
* Extends `Check` String extension to support finding multiple email addresses in a string
* Adds method extension to support Markdown and strip Markdown

**Additional**
* I ran _"Optimize imports"_ in Android Studio, which leads to changes in different files whose doesn't belong to the issue.

### CODE REVIEW

File: lib/src/utils/dialog_builder.dart [hunk 1]
```diff
@@ -50,16 +50,15 @@ import 'package:ox_coi/src/navigation/navigation.dart';
 
 import 'keyMapping.dart';
 
-showNavigatableDialog({
-  @required BuildContext context,
+showNavigatableDialog({@required BuildContext context,
   @required Widget dialog,
   @required Navigatable navigatable,
   Navigatable previousNavigatable,
-  barrierDismissible = true,
```
Comment (Boehrsi) [inline lib/src/utils/dialog_builder.dart]: No real problem, but please let's keep the `, ` at the end of the last item. If we do this adding a new entry at the end only affects one line and everything is easily readable in a diff. If we remove this we always add one `,` in the one line and the content + parenthesis in the other line.

Comment (phranck) [reply lib/src/utils/dialog_builder.dart]: I guess it's a habit of Swift development. It's not allowed there. 🙈

File: pubspec.yaml [hunk 2]
```diff
@@ -29,12 +29,14 @@ dependencies:
...
   flutter_html: ^0.11.1
   flutter_local_notifications: ^0.9.1+3
+  flutter_markdown: ^0.3.2
   flutter_sound:
     git:
       url: https://github.com/open-xchange/flutter_sound
   fluttertoast: ^3.1.3
   gettext: ^1.1.0+2
   gettext_parser: ^0.1.0+1
+  html2md:
```
Comment (Boehrsi) [inline pubspec.yaml]: We should add a specific version here.

Comment (phranck) [reply pubspec.yaml]: But they don't have any. See [project site](https://github.com/jarontai/html2md).

File: lib/src/message/message_builder.dart [hunk 3]
```diff
@@ -110,16 +113,26 @@ class MessageMaterial extends StatelessWidget {
...
+        data: markdown,
+        onTapLink: (url) {
+          _launch(url: url);
+        },
       ),
     );
   }
 }
 
+Future<void> _launch({@required String url}) async {
```
Comment (Boehrsi) [inline lib/src/message/message_builder.dart]: :+1: we should apply this logic + some sort of error toast / snackbar everywhere. I will create a ticket for that.

Comment (phranck) [reply lib/src/message/message_builder.dart]: Which logic do you mean? What is so special in this case...? 🤔

Comment (Boehrsi) [reply lib/src/message/message_builder.dart]: Checking for `canLaunch`, not that special, but I know some places where we just call launch without checking ;)

File: pubspec.yaml (hunk 2 above)
Comment (Boehrsi) [reply pubspec.yaml]: For this checking pub.dev is always the right source:
https://pub.dev/packages/html2md
html2md 0.5.1

**Pull request merged**